plugins/search-react/src/components/SearchResultGroup/SearchResultGroup.tsx

```diff
@@ -355,9 +366,9 @@ export type SearchResultGroupLayoutProps<FilterOption> = ListProps & {
  * @param props - See {@link SearchResultGroupLayoutProps}.
  * @public
  */
-export function SearchResultGroupLayout<FilterOption>(
-  props: SearchResultGroupLayoutProps<FilterOption>,
-) {
+export function SearchResultGroupLayout<T extends FilterOption>(
```

Comment:
Not sure I understand this change, it seems like `filterOptions` were previously unconstrained and could be any value, while the `renderFilterOption` would take care of converting it into JSX? I don't think we can constrain it to the new `FilterOption` as that would be a breaking change.

Reply:
To be honest I feel it would be enough to switch the default renderer to render `{String(filterOption)}` as the child. If one configures a set of filter options that aren't strings without overriding the default renderer, stuff ain't gonna look pretty either way 😅

Reply:
@Rugvip Okay, so you're saying that we should write this component & related logic assuming that the default option type is `string | number`, and allow the user to extend & provide their own rendering logic if they're using another type?

Reply:
> Therefore, it shouldn't break many existing usages, since filterOptions aren't likely to be objects in the current implementation.

FYI, @Rugvip and @RobotSail, this is incorrect. There is an example using both a string array and an object array.
[String array](https://github.com/backstage/backstage/blob/master/plugins/search-react/src/components/SearchResultGroup/SearchResultGroup.test.tsx#L275)
[Object array](https://github.com/backstage/backstage/blob/master/plugins/search-react/src/components/SearchResultGroup/SearchResultGroup.stories.tsx#L200-L209)

Reply:
Yep but the object array option provides s custom render function too. Without it you'd end up with broken option items